code/modules/mob/living/carbon/human/human.dm

```diff
@@ -430,39 +430,45 @@
 		else
 			return if_no_id
 
-//gets assignment from ID or ID inside PDA or PDA itself
-//Useful when player do something with computers
+//gets assignment from ID, PDA, Wallet, etc.
+//This should not be relied on for authentication, because PDAs show their owner's job, even if an ID is not inserted
 /mob/living/carbon/human/proc/get_assignment(if_no_id = "No id", if_no_job = "No job")
-	var/obj/item/pda/pda = wear_id
-	var/obj/item/card/id/id = wear_id
-	if(istype(pda))
-		if(pda.id && istype(pda.id, /obj/item/card/id))
-			. = pda.id.assignment
-		else
-			. = pda.ownjob
-	else if(istype(id))
-		. = id.assignment
-	else
+	if(!wear_id)
 		return if_no_id
-	if(!.)
-		. = if_no_job
-	return
+	var/obj/item/card/id/id = wear_id.GetID()
+	if(istype(id)) // Make sure its actually an ID
+		if(!id.assignment)
+			return if_no_job
+		return id.assignment
+
+	if(is_pda(wear_id))
+		var/obj/item/pda/pda = wear_id
+		return pda.ownjob
 
-//gets name from ID or ID inside PDA or PDA itself
-//Useful when player do something with computers
+	return if_no_id
+
+//gets name from ID, PDA, Wallet, etc.
+//This should not be relied on for authentication, because PDAs show their owner's name, even if an ID is not inserted
 /mob/living/carbon/human/proc/get_authentification_name(if_no_id = "Unknown")
-	var/obj/item/pda/pda = wear_id
-	var/obj/item/card/id/id = wear_id
-	if(istype(pda))
-		if(pda.id)
-			. = pda.id.registered_name
-		else
-			. = pda.owner
-	else if(istype(id))
-		. = id.registered_name
-	else
+	if(!wear_id)
 		return if_no_id
-	return
+	var/obj/item/card/id/id = wear_id.GetID()
+	if(istype(id) && id.registered_name)
+		return id.registered_name
+
+	if(is_pda(wear_id))
+		var/obj/item/pda/pda = wear_id
+		return pda.owner
+
+	return if_no_id
+
+/mob/living/carbon/human/proc/GetIdCard(mob/living/carbon/human/H)
```

Comment:
`snake_case`? 👉 👈

Reply:
snake_cased, but this also makes it a subtype of an already existing proc on mob called /get_id_card, so I just made this the human child proc of it